### ISSUE TITLE
Empty Javascript File When Building

### ISSUE THREAD

**Issue by preston-mro** [parcel-bundler/parcel#7412]
# 🐛 bug report

In my package.json:
```
"dev": "parcel parcel/index.scss parcel/index.js --dist-dir assets --no-source-maps",
"build": "parcel build parcel/index.scss parcel/index.js --dist-dir assets --no-source-maps"
```

_dev_ works. I get the expected results, which is my two index files are dropped into the root of my _assets_ folder and both are populated with code. I can see my Sass and JS coming in from the separate files.
_build_ will also output both index files, however, the js file is empty. The css file is populated.

No errors. Build completes.

All other settings are default.

Let me know if you need any more info. Thanks for any help.

## 🤔 Expected Behavior
index.js has code in it.

## 😯 Current Behavior
index.js is empty.

### How has this issue affected you? What are you trying to accomplish?
I am a Shopify developer and I want to set up a build tool for JS and Sass.

## 🌍 Your Environment
I'm on a Win10 laptop running the latest node and npm.

**Comment by astegmaier** [comment]
This issue was also posted on stackoverflow, and I managed [to resolve it there](https://stackoverflow.com/a/70284901/4739687). Feel free to close.

The problem was that the entire bundle was getting tree-shaken away - because it was intended to expose the contents as a global variables - which were not used in the bundle itself. We managed to find a workaround, but this process will become simpler once #7312 is fixed.
